pkg/check/util.go

```
@@ -334,26 +340,53 @@ func GetVCenter(checkContext *CheckContext, node *v1.Node) (*VCenter, error) {
+		if len(checkContext.PlatformSpec.FailureDomains) > 0 {
+			klog.V(2).Infof("Returning first FD defined in platform spec.")
+			fd := checkContext.PlatformSpec.FailureDomains[0]
+			vc := checkContext.VCenters[fd.Server]
+			if vc == nil {
+				return nil, fmt.Errorf("vCenter %v not found", fd.Server)
+			}
+			return vc, nil
+		} else {
+			klog.V(2).Infof("There are no failure domains in platform spec. Returning first vCenter from context.")
+			if len(checkContext.VCenters) > 1 {
+				return nil, fmt.Errorf("invalid number of configured vCenters detected: %d.  Expected only 1.", len(checkContext.VCenters))
```

golangci-lint (2.12.2)

[error] ST1005: error strings should not end with punctuation or newlines

(staticcheck)